Dshop/apps/users/api_views.py

```diff
@@ -0,0 +1,44 @@
+from django.contrib.auth import get_user_model
+from django.contrib.auth import login as django_login
+from rest_framework import viewsets, mixins, status
+from rest_framework.authtoken.models import Token
+from rest_framework.generics import GenericAPIView
+from rest_framework.permissions import AllowAny
+from rest_framework.response import Response
+
+from .models import CustomUser
+from .serializers import RegistrationSerializer, LoginSerializer, TokenSerializer
+
+User = get_user_model()
+
+
+class RegistrationViewSet(mixins.CreateModelMixin, viewsets.GenericViewSet):
+    queryset = User.objects.all()
+    permission_classes = (AllowAny,)
+    throttle_scope = 'registration'
+    serializer_class = RegistrationSerializer
+
+    def perform_create(self, serializer):
+        user = serializer.save()
+        custom_user = CustomUser(user=user)
+        custom_user.save()
+
+
+class LoginView(GenericAPIView):
+    permission_classes = (AllowAny,)
+    serializer_class = LoginSerializer
+    throttle_scope = 'login'
+
+    def post(self, request):

```

Comment:
This returns not enough data. User data will be required for example to show a welcome message:
Hi Jacek, welcome back

DRF shows:
```
from rest_framework.authtoken.views import ObtainAuthToken
from rest_framework.authtoken.models import Token
from rest_framework.response import Response

class CustomAuthToken(ObtainAuthToken):

    def post(self, request, *args, **kwargs):
        serializer = self.serializer_class(data=request.data,
                                           context={'request': request})
        serializer.is_valid(raise_exception=True)
        user = serializer.validated_data['user']
        token, created = Token.objects.get_or_create(user=user)
        return Response({
            'token': token.key,
            'user_id': user.pk,
            'email': user.email
        })
```
https://www.django-rest-framework.org/api-guide/authentication/